sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala

```diff
@@ -159,7 +159,6 @@ object FileFormatWriter extends Logging {
     val actualOrdering = writeFilesOpt.map(_.child)
       .getOrElse(materializeAdaptiveSparkPlan(plan))
       .outputOrdering.map(_.child)
-    val orderingMatched = V1WritesUtils.isOrderingMatched(requiredOrdering, actualOrdering)
```

Comment:
why this line doesn't work anymore?

Reply:
We do not add the sort in rule `V1Writes` since the ordering matches. So the actual ordering is the user-specified which does not rewrite the attribute using the new query that wrapped with `Empty2Null`. But the required ordering has rewritten in rule `V1Writes`.